openassessment/xblock/test/test_grade_explanation.py

```diff
@@ -0,0 +1,263 @@
+    }
+
+    assessment_score_priority = (
+        ('self', 'peer'),
+        ('peer', 'self')
+    )
+
+    @scenario('data/grade_scenario_self_only.xml', user_id='Greggs')
+    @data(*assessment_score_priority)
+    def test_render_grade_explanation_self_only(self, xblock, assessment_score_priority):
+        settings.ORA2_ASSESSMENT_SCORE_PRIORITY = assessment_score_priority
+        self.create_submission_and_assessments(
+            xblock, self.SUBMISSION, [], [], SELF_ASSESSMENT,
+            waiting_for_peer=True
+        )
+        resp = self.request(xblock, 'render_grade', json.dumps(dict()))
+
+        self.assertIn(self.second_sentences_options["self"], resp.decode('utf-8'))
+
+    @scenario('data/grade_scenario_staff_only.xml', user_id='Bernard')
+    @data(*assessment_score_priority)
+    def test_render_explanation_grade_staff_only(self, xblock, assessment_score_priority):
+        settings.ORA2_ASSESSMENT_SCORE_PRIORITY = assessment_score_priority
+        student_item = xblock.get_student_item_dict()
+
+        submission = xblock.create_submission(student_item, self.SUBMISSION)
+        self.submit_staff_assessment(xblock, submission, STAFF_GOOD_ASSESSMENT)
+        resp = self.request(xblock, 'render_grade', json.dumps(dict()))
+
+        self.assertIn(self.second_sentences_options["staff"], resp.decode('utf-8'))
+
+    @scenario('data/grade_scenario_peer_only.xml', user_id='Bernard')
+    @data(*assessment_score_priority)
+    def test_render_grade_explanation_peer_only(self, xblock, assessment_score_priority):
+        settings.ORA2_ASSESSMENT_SCORE_PRIORITY = assessment_score_priority
+        self.create_submission_and_assessments(
+            xblock, self.SUBMISSION, self.PEERS, PEER_ASSESSMENTS, None
+        )
+        resp = self.request(xblock, 'render_grade', json.dumps(dict()))
+
+        self.assertIn(self.second_sentences_options["peer"], resp.decode('utf-8'))
+
+    @scenario('data/grade_scenario.xml', user_id='Bernard')
+    @data(*assessment_score_priority)
+    def test_render_grade_explanation_self_and_peer(self, xblock, assessment_score_priority):
+        settings.ORA2_ASSESSMENT_SCORE_PRIORITY = assessment_score_priority
+        self.create_submission_and_assessments(
+            xblock, self.SUBMISSION, self.PEERS, PEER_ASSESSMENTS, SELF_ASSESSMENT,
+        )
+        resp = self.request(xblock, 'render_grade', json.dumps(dict()))
+
+        if assessment_score_priority.index('self') < assessment_score_priority.index('peer'):
+            self.assertIn(self.second_sentences_options["self"], resp.decode('utf-8'))
+        else:
+            self.assertIn(self.second_sentences_options["peer"], resp.decode('utf-8'))
+
+    @scenario('data/grade_scenario_self_staff.xml', user_id='Bernard')
+    @data(*assessment_score_priority)
+    def test_render_grade_explanation_self_and_staff(self, xblock, assessment_score_priority):
```

Comment:
Same as the previous comment here and in all other tests that include a `staff` grade, change the assessment priority if the response is always the same? `staff` will always prevail, no matter which other priorities exist, so these extra tests are not really usefull.

Reply:
See me previous comment.